File: js/packages/app/src/__tests__/features/object-references/internal/polywrap/internal-references-example/src/polywrap/WrapManifest.ts
export namespace WrapManifest {
  export namespace Internal {
    export enum GlobalFunction {
            
    }

    export enum Class {
      
      
      
      
    }

    export namespace Classes {
      
      
      
      
    }
  }

  export namespace External {
    export enum GlobalFunction {
      
      TestReturnReference = 0,
      
    }

    export enum Class {
      
      TestInternalClass = 0,
      
      
      TestObjectGetter = 1,
      
    }

    export namespace Classes {
      
      export enum TestInternalClassMethod {
        Create = 0,
        TestInstanceMethod = 1,
      }
      
      
      export enum TestObjectGetterMethod {
        Create = 0,
        TestInstanceMethod = 1,
        TestStaticMethod = 2,
      }
      
    }
  }
}

File: rust/app/src/wrappers/external_references/src/polywrap/wrap/wrap_manifest.rs
pub mod internal {
    pub enum GlobalFunction {
        TestReceiveReference = 0,
        TestInvokeExternalGlobalFunction = 1,
        TestInvokeExternalStaticMethod = 2,
        TestInvokeExternalInstanceMethod = 3,
    }

    pub enum Class {
        TestObjectGetter = 0,
    }

    pub mod classes {
        pub enum TestObjectGetterMethod {
        Create = 0,
        TestInstanceReceiveReference = 1,
        TestStaticReceiveReference = 2,
        }
    }
    }

pub mod external {
    pub enum GlobalFunction {
        TestExternalGlobalFunction = 4,
    }

    pub enum Class {
        TestExternalClass = 1,
    }

    pub mod classes {
        pub enum TestExternalClassMethod {
        Create = 0,
        TestInstanceMethod = 1,
        TestStaticMethod = 2,
        }
    }
}
